### PR TITLE
Add chiLife recipe

### PR DESCRIPTION
**Pull request by mtessmer**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mtessmer** [comment]
@conda-forge-admin, please ping conda-forge/help-python PR is ready for review

File: recipes/chilife/LICENSE [hunk 1]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (ocefpaf) [inline recipes/chilife/LICENSE:1]: This file is in the tarball, no need to add it here.

File: recipes/chilife/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{% set name = "chilife" %}
+{% set version = "1.0.0.dev6" %}
```
Comment (ocefpaf) [inline recipes/chilife/meta.yaml:2]: This one falls under https://github.com/conda-forge/cfep/blob/main/cfep-05.md

File: recipes/chilife/meta.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {% set name = "chilife" %}
-{% set version = "1.0.0.dev6" %}
```
Comment (ocefpaf) [inline recipes/chilife/meta.yaml]: Note that you can have the 1.0 dev version in a dev channel. The instructions should be in the CFEP-05 doc I linked in the previous comment. You can do that in the feedstock later if you wish.

**Pull request merged**